src/client/Microsoft.Identity.Client/UI/AuthorizationResult.cs

```diff
@@ -43,44 +43,50 @@ public static AuthorizationResult FromUri(string webAuthenticationResult)
                    MsalErrorMessage.AuthorizationServerInvalidResponse);
             }
 
-            Dictionary<string, string> uriParams = CoreHelpers.ParseKeyValueList(resultData.Substring(1), '&',
-                true, null);
+            Dictionary<string, string> uriParams = CoreHelpers.ParseKeyValueList(
+                resultData.Substring(1), '&', true, null);
 
-            if (uriParams.ContainsKey(TokenResponseClaim.Error))
+            var result = FromParsedValues(uriParams);
+
+            if (uriParams.ContainsKey(TokenResponseClaim.Code))
             {
-                return FromStatus(AuthorizationStatus.ProtocolError,
-                    uriParams[TokenResponseClaim.Error],
-                    uriParams.ContainsKey(TokenResponseClaim.ErrorDescription)
-                            ? uriParams[TokenResponseClaim.ErrorDescription]
-                        : null);
+                result.Code = uriParams[TokenResponseClaim.Code];
             }
-
-            var result = new AuthorizationResult();
-            result.Status = AuthorizationStatus.Success;
-
-            if (uriParams.ContainsKey(OAuth2Parameter.State))
+            else if (webAuthenticationResult.StartsWith("msauth://", StringComparison.OrdinalIgnoreCase))
             {
-                result.State = uriParams[OAuth2Parameter.State];
+                result.Code = webAuthenticationResult;
             }
-
-            if (uriParams.ContainsKey(TokenResponseClaim.CloudInstanceHost))
+            else
             {
-                result.CloudInstanceHost = uriParams[TokenResponseClaim.CloudInstanceHost];
+                return FromStatus(
+                   AuthorizationStatus.UnknownError,
+                   MsalError.AuthenticationFailed,
+                   MsalErrorMessage.AuthorizationServerInvalidResponse);
             }
 
-            if (uriParams.ContainsKey(TokenResponseClaim.ClientInfo))
+            return result;
+        }
+#if !UAP10_0 && !NETSTANDARD1_3
+        public static AuthorizationResult FromPostData(byte[] postData)
```

Comment:
This and `AppendOrReplaceQueryParameter` are probably the only things you can unit test...

Reply:
Added tests.